### PR TITLE
Migrate publishing plugin

### LINKED ISSUE #484: Migrate Publishing Plugin to bintray/gradle-bintray-plugin instead
# Refactoring
Replacing novoda/bintray-release with bintray/gradle-bintray-plugin
## Description

As mentioned at the upstream repo, as of now (26 Oct) novoda/bintray-release has an issue that cannot support gradle 4.5 or higher. Even though it is an awesome plugin (so easy to use), the world that we are moving is too fast and staying behind on gradle version is not that fun 🙅‍♂️ . 

## Files to process
- [ ] build.gradle.kts

## Why this is necessary**
We want to stay on top of the latest Gradle plugin so that we can use the newer functionalities.

## Additional context
N/A

### PR DESCRIPTION
**Pull request by kittinunf**
## Description

I am working on migrating to the better plugin so that we don't have to stuck with gradle 4.4 any longer.

Fixes #484 

## Type of change

Check all that apply

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (a change which changes the current internal or external interface)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Manually ... 😢 😭 

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new compiler warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Inspect the bytecode viewer, including reasoning why

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -114,29 +117,66 @@ subprojects {
...
+            })
+        }
+
+        if (project.hasProperty("android").not()) {
+            val sourcesJar by tasks.registering(Jar::class) {
+                classifier = "sources"
+                from(sourceSets["main"].allSource)
+            }
+
+            val doc by tasks.registering(Javadoc::class) {
```
Comment (NikkyAI) [inline build.gradle.kts]: this can be changed to `creating` and it doesn ot have to be accessed with `.get()`

File: build.gradle.kts [hunk 2]
```diff
@@ -114,29 +117,66 @@ subprojects {
...
-
-        tasks.withType<Javadoc>().all { enabled = false }
     }
 
     if (!isSample) {
         apply {
-            plugin(BintrayRelease.plugin)
-            plugin(Ktlint.plugin)
-        }
+            plugin("org.gradle.maven-publish")
```
Comment (NikkyAI) [inline build.gradle.kts]: in the long form necessary ?
would ´ plugin("maven-publish")´ not work?

**Comment by NikkyAI** [comment]
this looks good, i trust it works as well ?

File: build.gradle.kts (hunk 2 above)
Comment (iNoles) [reply build.gradle.kts]: Odd, you have `maven-publish` in the main plugins block

Comment (kittinunf) [reply build.gradle.kts]: I need 2 places or it will not be able to sync. Do you have a clue why though?

**Comment by kittinunf** [comment]
I don't think it is ... I am working on the Android project related support.

File: build.gradle.kts (hunk 1 above)
Comment (kittinunf) [reply build.gradle.kts]: I think I need to use the `.get()` still after changed to `creating` ...

**Comment by kittinunf** [comment]
@NikkyAI @iNoles One more check, sir!

**Comment by SleeplessByte** [comment]
LGTM, I've re-invited @iNoles and @NikkyAI as they have waaay more knowledge on this gradle stuff.

**Comment by iNoles** [comment]
I guess it work correctly. It kinda sucks to test it by pushing it.

File: build.gradle.kts [hunk 3]
```diff
@@ -107,36 +118,98 @@ subprojects {
...
-            website = "https://github.com/kittinunf/Fuel"
-        }
-
         configure<KotlinterExtension> {
             reporters = arrayOf(ReporterType.plain.name, ReporterType.checkstyle.name)
         }
+
+        version = Fuel.publishVersion
+        group = Fuel.groupId
+        configure<BintrayExtension> {
```
Comment (NikkyAI) [inline build.gradle.kts]: not sure if this could be replaced with 
```
bintray {
```

**Comment by NikkyAI** [comment]
the high level of nesting is a shame.. and we can probably later move the worst into buildSrc or at least chop it up into smaller functions

File: build.gradle.kts (hunk 3 above)
Comment (kittinunf) [reply build.gradle.kts]: Oh really? I will try!

Comment (kittinunf) [reply build.gradle.kts]: 🙅‍♂️ I couldn't ... >_<

**Comment by SleeplessByte** [comment]
@kittinunf I think you can merge when ready 💯

File: build.gradle.kts (hunk 3 above)
Comment (iNoles) [reply build.gradle.kts]: https://github.com/bintray/gradle-bintray-plugin/issues/258 provides example that should work.

**Comment by kittinunf** [comment]
> the high level of nesting is a shame.. and we can probably later move the worst into buildSrc or at least chop it up into smaller functions

Let's do that later then. I think I am more or less done with fighting with build gradle for now.

**Pull request merged**